social-network-api/src/main/java/fs/socialnetworkapi/service/PasswordResetService.java

```diff
@@ -2,21 +2,17 @@
 
 import fs.socialnetworkapi.component.AppLink;
 import fs.socialnetworkapi.entity.User;
-import lombok.RequiredArgsConstructor;
-import org.springframework.beans.factory.annotation.Autowired;
+import lombok.AllArgsConstructor;
 import org.springframework.stereotype.Service;
 
 import java.util.Optional;
 import java.util.UUID;
 
 @Service
-@RequiredArgsConstructor
+@AllArgsConstructor
```

Comment:
Why not RequiredArgsConstructor?

Reply:
Why is it a big difference?

Reply:
If you will use RequiredArgsConstructor it will use only the final fields in the constructor. In this case, you can easily inject the value of the application link using annotation.